File: src/Lesson4/java/Ex3.java
package Lesson4.java;
import java.util.Scanner;

public class Ex3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Welcome to the Berland to Birland translation verification program");
        System.out.println("Enter word: ");

        String s = scanner.nextLine();
        String t = scanner.nextLine();

        if (isReverse(s, t)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    static boolean isReverse(String s, String t) {
        return new StringBuilder(s).reverse().toString().equals(t);
    }

}

File: src/Lesson4/forkotlin/Ex3.kt
package Lesson4.forkotlin

fun main() {
    println("Welcome to the Berland to Birland translation verification program")
    println("Enter word: ")
    val s = readln()
    val t = readln()

    if (s.reversed() == t) {
        println("YES")
    } else {
        println("NO")
    }
}